gnd/src/main/java/com/google/android/gnd/repository/OfflineBaseMapRepository.java

```diff
@@ -256,4 +258,55 @@ public Completable deleteArea(String offlineAreaId) {
                     .andThen(localDataStore.deleteTileByUrl(tile)))
         .andThen(localDataStore.deleteOfflineArea(offlineAreaId));
   }
+
+  /**
+   * Retrieve all tile sources from a GeoJSON basemap specification, regardless of their
+   * coordinates.
+   */
+  public Single<ImmutableList<TileSource>> getAllActiveTileSources() {
+    // TODO: Only load tiles for basemap sources that are "active" (we will implement support for
+    // turing sources on and off).
+    return projectRepository
+        .getProjectLoadingState()
+        .compose(Loadable::values)
+        .map(Project::getOfflineBaseMapSources)
+        .doOnError(
+            throwable -> Timber.e(throwable, "No basemap sources specified for the active project"))
+        .map(ImmutableList::asList)
+        .flatMap(Flowable::fromIterable)
+        .firstOrError()
+        .map(this::identifyAndHandleSource)
+        .doOnError(
+            throwable ->
+                Timber.e(throwable, "Couldn't retrieve basemap sources for the active project"));
+  }
+
+  private ImmutableList<TileSource> identifyAndHandleSource(
```

Comment:
Yeah, tbh I find the mutual existence of TileSource and BasemapSource confusing. Originally we distinguished them since a footprints file contains N tile sources (since it specifies explicit URL resources for its tiles).

I think unifying this all into a `BaseMapSource` is probably best.